components/sync15/src/state.rs

```diff
@@ -756,6 +789,22 @@ mod tests {
             default: KeyBundle::new_random().unwrap(),
             collections: HashMap::new(),
         };
+        let mut mg = MetaGlobalRecord {
+            sync_id: "syncIDAAAAAA".into(),
+            storage_version: 5usize,
+            engines: vec![(
+                "bookmarks",
+                MetaGlobalEngine {
+                    version: 1usize,
+                    sync_id: "syncIDBBBBBB".into(),
+                },
+            )]
+            .into_iter()
+            .map(|(key, value)| (key.to_owned(), value))
+            .collect(),
+            declined: vec![],
+        };
+        fixup_meta_global(&mut mg);
```

Comment:
Hmm, should we let the state machine fix this up for us as part of the test, and verify we've uploaded a new record with the engines?